### PR TITLE
Add static z3 build from deps PPA to testing docker image.

### PR DESCRIPTION
**Pull request by ekpyron**

### CODE REVIEW

File: .circleci/docker/Dockerfile.ubuntu1904 [hunk 1]
```diff
@@ -34,7 +36,7 @@ RUN set -ex; \
 		libboost-regex-dev libboost-filesystem-dev libboost-test-dev libboost-system-dev \
 		libboost-program-options-dev \
 		libjsoncpp-dev \
-		llvm-8-dev libcvc4-dev libleveldb1d \
+		llvm-8-dev libcvc4-dev libz3-static-dev libleveldb1d \
```
Comment (ekpyron) [inline .circleci/docker/Dockerfile.ubuntu1904:40]: If the static Z3 build turns out to work, we might want to remove libcvc4-dev from the image.

File: .circleci/config.yml [hunk 2]
```diff
@@ -181,7 +181,7 @@ jobs:
             pip install --user codespell
       - run:
           name: Check spelling
-          command: ~/.local/bin/codespell -S "*.enc,.git" -I ./scripts/codespell_whitelist.txt
+          command: ~/.local/bin/codespell -S "*.enc,.git,Dockerfile*" -I ./scripts/codespell_whitelist.txt
```
Comment (ekpyron) [inline .circleci/config.yml:184]: Any other or better suggestions? It complained about "keyserver": https://circleci.com/gh/ethereum/solidity/158189

Comment (christianparpart) [reply .circleci/config.yml:184]: I think this is the only way to do it AIUI, as codespell doesn't treat Dockerfile syntax differently. (for example we might still want to spell-check comments within this file).

File: .circleci/docker/Dockerfile.ubuntu1904 [hunk 3]
```diff
@@ -26,6 +26,8 @@ FROM buildpack-deps:disco AS base
 ARG DEBIAN_FRONTEND=noninteractive
 
 RUN set -ex; \
+	echo "deb http://ppa.launchpad.net/ethereum/cpp-build-deps/ubuntu disco main" >> /etc/apt/sources.list ; \
```
Comment (christianparpart) [inline .circleci/docker/Dockerfile.ubuntu1904]:
```suggestion
	echo "deb http://ppa.launchpad.net/ethereum/cpp-build-deps/ubuntu $(lsb_release -cs) main" >> /etc/apt/sources.list ; \
```
I think it's not bad to make this line a little more generic, to make future upgrades of ubuntu docker image versions less error prone. What do you think?

Comment (ekpyron) [reply .circleci/docker/Dockerfile.ubuntu1904]: I like it - but sadly ``lsb_release`` won't be available here (not part of buildpack-deps:disco itself and no sense in installing it only for this).
Maybe ``source /etc/lsb-release && echo "deb http://ppa.launchpad.net/ethereum/cpp-build-deps/ubuntu $DISTRIB_CODENAME main" >> /etc/apt/sources.list ; \``?

Comment (ekpyron) [reply .circleci/docker/Dockerfile.ubuntu1904]: I'll change it in the PR - let me know if it works.

File: .circleci/config.yml (hunk 2 above)
Comment (ekpyron) [reply .circleci/config.yml:184]: That's what I thought, yep.

File: .circleci/docker/Dockerfile.ubuntu1904 (hunk 3 above)
Comment (christianparpart) [reply .circleci/docker/Dockerfile.ubuntu1904]: :-1: sadly.

**Review by christianparpart — APPROVED** [review]
Tested locally and also uploaded the new docker image already.

**Pull request merged**